NBitcoin.Tests/NetworkAddressTests.cs

```diff
@@ -0,0 +1,545 @@
+using System.IO;
+using System.Linq;
+using System.Threading;
+using NBitcoin;
+using Xunit;
+using System;
+using NBitcoin.Protocol;
+using System.Net;
+using NBitcoin.DataEncoders;
+
+namespace NBitcoin.Tests
+{
+	public class NetworkAddressTests
+	{
+		[Fact]
+		[Trait("UnitTest", "UnitTest")]
+		public void CanParseSpecialAddresses()
+		{
+			var addr = new NetworkAddress();
+						
+			// TORv2
+			Assert.True(addr.SetSpecial("6hzph5hv6337r6p2.onion"));
```

Comment:
Would it make sense to a constructor that allows to pass an onion address or a factory method if not?

Reply:
It could make sense but I prefer to keep this close to the core implementation because in that way it is easier to compare and make changes in the future.